bip-vaults.mediawiki

```diff
@@ -0,0 +1,745 @@
+<code>OP_UNVAULT</code>, that add consensus support for a specialized covenant. 
+These opcodes allow users to enforce a delay period before designated coins may
+be spent to an arbitrary destination, with the exception of a prespecified
+"recovery" path. At any time prior to final withdrawal, the coins can be spent to
+the prespecified path.
+
+
+=== Motivation ===
+
+The hazard of custodying Bitcoin is well known. Users of Bitcoin must go to
+significant effort to secure their private keys, and hope that once provisioned
+their custody system does not yield to any number of evolving and
+persistent threats. Users have little means to intervene once compromise is
+detected. This proposal introduces a mechanism that significantly
+mitigates the worst-case outcome of key compromise: coin loss.
+
+Introducing a way to intervene during unexpected spends allows users to
+incorporate highly secure key storage methods or unusual fallback strategies
+that are only exercised in the worst case, and which may otherwise be
+operationally prohibitive. The goal of this proposal is to make this strategy
+usable for custodians of any size with minimal complication.
+
+==== Example uses ====
+
+An individual custodying Bitcoin uses the common "single signature and
+passphrase" configuration with a hardware wallet. They are concerned about the
+risk associated with relying on a single manufacturer for key management as
+well as physical access to the hardware, so they generate a new key that is
+highly secure, but would be impractical for daily use. For example the key
+could be generated in some analog fashion, or on an old computer (with added
+entropy) that is then destroyed, with the private key replicated only in paper
+form. Or the key could be a 2-of-3 multisig using devices from different
+manufacturers. Perhaps the key is geographically distributed.
+
+This individual can use <code>OP_VAULT</code> to make use of the highly secure
+key as the unlikely recovery path, while using their existing signing procedure
+as the withdrawal trigger key, with a configured spend delay of 1 day. They can
+run software on their mobile device that monitors the blockchain for spends of
+the vault outpoints.
+
+If the vaulted coins move in an unexpected way, the user can immediately sweep
+them to the highly secure recovery path, but spending the coins on a daily
+basis works in the same way it did prior to vaulting - aside from the spend
+delay.
+
+The recovery key could be any number of things: a 2-of-3 multisig with keys
+that live on different devices, a 3-of-5 with socially distributed keys, a
+Taproot construction that incorporates one of these methods along with a
+time-delayed fallback to an "easier" recovery method, in case the highly secure
+key winds up being ''too'' highly secure.
+
+Institutional custodians of Bitcoin would likely use vaults in similar fashion.
+
+===== Avoiding the $5 wrench attack =====
```

Comment:
Since the section already mentions the $5 wrench attack, the title itself could be something more descriptive, such as "Provably non-bypassable timelocks".